server/src/user/user.controller.ts

```diff
@@ -16,4 +17,9 @@ export class UserController {
   getUsers(): Promise<User[]> {
     return this.userService.getUsers();
   }
+
+  @Patch('/:id/access')
```

Comment:
Missed this. I've also added Admin based auth! Refer the user fetch route and add it accordingly!

Reply:
Hey Karthikk, I see that the Get() method has the @Roles(UserRoles.ADMIN) decorator. What does it do aside from setting the roles metadata to Admin? Is this on its own enough to validate that it's an admin? If so, how?
I see there's also a GetUser parameter decorator that gets the logged user and their roles. Should I instead do that and then do a simple check like "if roles ==== UserRoles.Admin -> run the function, else do nothing"?
Thanks in advance!

Reply:
Check roles.guard.ts for better understanding!